sds/src/match_validation/validator_utils.rs

```diff
@@ -8,11 +10,47 @@ use super::{
 pub fn new_match_validator_from_type(config: &MatchValidationType) -> Box<dyn MatchValidator> {
     match config {
         MatchValidationType::Aws(aws_config) => match aws_config {
-            AwsType::AwsSecret(_) => Box::new(AwsValidator::new()),
+            AwsType::AwsSecret(aws_config) => Box::new(AwsValidator::new(aws_config.clone())),
             _ => panic!("This aws type shall not be used to create a validator"),
         },
         MatchValidationType::CustomHttp(http_config) => {
             Box::new(HttpValidator::new(http_config.clone()))
         }
     }
 }
+
+pub fn generate_aws_headers_and_body(
+    datetime: &DateTime<Utc>,
+    endpoint: &str,
+    aws_id: &str,
+    aws_secret: &str,
+) -> (String, reqwest::header::HeaderMap) {
+    let mut headers = reqwest::header::HeaderMap::new();
+    let datetime_str = datetime.format("%Y%m%dT%H%M%SZ").to_string();
+    headers.insert("X-Amz-Date", datetime_str.parse().unwrap());
+    headers.insert("Accept-Encoding", "identity".parse().unwrap());
+    headers.insert(
+        "Content-Type",
+        "application/x-www-form-urlencoded; charset=utf-8"
+            .parse()
+            .unwrap(),
+    );
+    headers.insert("host", "sts.us-east-1.amazonaws.com".parse().unwrap());
+
+    let body = "Action=GetCallerIdentity&Version=2011-06-15".to_string();
+    let s = aws_sign_v4::AwsSign::new(
+        "POST",
+        endpoint,
+        &datetime,
+        &headers,
+        "us-east-1", // default region
```

Comment:
Does this need to be customizable, or is `us-east-1` always valid?

Reply:
We are using a default sts endpoint that is hosted in us-east-1 
It should solved all request from all endpoint. But this is a good point I haven't tested with credentials from other DC. I will do that as a follow up

Reply:
We are relying on a global sts endpoint that is hosted in us-east-1
All IAM account are global so I guess since that works with the test credential it should be good